### ISSUE TITLE
woocommerce_product_get_tax_class no longer runs on admin

### ISSUE THREAD

**Issue by webthread** [woocommerce/woocommerce#16368]
Prior to 3.X `woocommerce_product_tax_class` would run on admin generated orders.
Now neither the above or `woocommerce_product_get_tax_class` run on the admin side at all. This has completely broken my wholesale filters as they only run on the frontend.

Is this a deliberate change?
Any clues on a way around it?

Thanks
Nik

**Comment by mikejolley** [comment]
@webthread It does if you're talking about the same piece of code. `add_product` method for orders uses `$product->get_tax_class()`:

https://github.com/woocommerce/woocommerce/blob/3522857d23051844bea742a37a1b0b242f7850af/includes/abstracts/abstract-wc-order.php#L879

So `woocommerce_product_get_tax_class` will run.

**Comment by webthread** [comment]
@mikejolley I can't work out why this function works in 2.X (with `woocommerce_product_tax_class`) for the frontend and backend generated orders. But in 3.X it doesn't fire for backend orders anymore?
```
function wc_diff_rate_for_user( $tax_class, $product ) {
        $user_id = get_current_user_id();
        $user = get_user_by( 'id', $user_id );
    if ( is_user_logged_in() &&
            in_array( 'row_trade', $user->roles ) ||
            in_array( 'eu_trade', $user->roles ) {
        $tax_class = 'Xero Rate';
    }
    return $tax_class;
}
add_filter( 'woocommerce_product_get_tax_class', 'wc_diff_rate_for_user', 1, 2 );
```

**Comment by webthread** [comment]
Thanks @mikejolley it seems even with the slug and settings the function won't run on the admin side. We are looking to see if WC_Customer set_is_vat_exempt() is another option.

This filter (or a similar one) working on the backend is vital for anyone with wholesale customers as at the moment as soon as you touch the order in the admin it adds tax according to the table settings, you can then delete the tax column on the order but pressing recalculate simply adds it back again. Would it be possible to consider a filter that can access the admin orders?

**Comment by mikejolley** [comment]
I tested this with the slug by var_dumping in the above function. It runs when the product is added to the order.

**Comment by webthread** [comment]
@mikejolley correct, I get the same behaviour. But if you need to add shipping or make other adjustments to the order the recalculate button ignores the filter. The save button is ok, just seems to be an issue with recalculate?

**Comment by webthread** [comment]
For me, not being able to have some control to override taxes in admin orders is a bug. If there is a different way to approach this in 3.X it would be great to know?

**Comment by mikejolley** [comment]
If you wanted to submit a PR adding VAT exception to customers for the backend that would be good, then WC could be left to do calculations.

**Comment by webthread** [comment]
Alrighty, I will discuss with my developer and see if we can find a solution. Do you have any comments on the best approach to this?

**Comment by mikejolley** [comment]
I presume an extra item of meta for customers. It can be added to the WC_Customer object + the customer data stores. Probably needs a checkbox on the user edit screen too.

**Comment by leendertvb** [comment]
Sorry, I know this an old thread, but for anyone stumbling on this question, please allow me to try and point you in a direction that might help you.

Please look into the action hook `woocommerce_order_item_after_calculate_taxes` as you can find in the function `WC_Order_Item ::calculate_taxes()`. See: https://plugins.trac.wordpress.org/browser/woocommerce/trunk/includes/class-wc-order-item.php#L215

In this action hook you have access to a `WC_Order_Item` object so you can easily access the function `get_order()` to get the full order and continue from there.

Example:
```
add_action( 'woocommerce_order_item_after_calculate_taxes', 'iside_order_item_after_calculate_taxes', 10, 2 );
function iside_order_item_after_calculate_taxes($wc_order_item, $calculate_tax_for = array()) {
    $wc_order = $wc_order_item->get_order();
    $product_tax_class = $calculate_tax_for['tax_class'];
    // do whatever you like :-)
}
```
